plugin/src/main/java/io/jenkins/plugins/casc/yaml/YamlSource.java

```diff
@@ -45,7 +49,17 @@ public String source() {
 
     public static final YamlReader<String> READ_FROM_URL = config -> {
         final URL url = URI.create(config).toURL();
-        return new InputStreamReader(url.openStream(), UTF_8);
+        URLConnection connection = url.openConnection();
```

Comment:
Nice feature, would be good if you could add a test?
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/test/java/io/jenkins/plugins/casc/yaml/YamlSourceTest.java

even just verify that the property is set correctly with the values you expect?

Reply:
> Nice feature, would be good if you could add a test?
> https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/test/java/io/jenkins/plugins/casc/yaml/YamlSourceTest.java
> 
> even just verify that the property is set correctly with the values you expect?

Hi Tim,

I can't find a way to test my changes as they are entirely depend on System environment variables.
I also reviewed the test that you referred me to and they just testing the YamlSource object's constructor and not 'READ_FROM_URL' method.

Do you have any suggestion how to test the change that I have made?

Thanks,
Matan.

Reply:
Possibly this?
https://stefanbirkner.github.io/system-rules/#ProvideSystemProperty

Reply:
There are plenty of examples how to test with environment variables in the existing tests :)

https://github.com/jenkinsci/configuration-as-code-plugin/blob/0dace8f8e5f5e85395c6e914dd5b321a26c61362/plugin/src/test/java/io/jenkins/plugins/casc/SecretSourceResolverTest.java#L16-L21

https://github.com/jenkinsci/configuration-as-code-plugin/blob/0dace8f8e5f5e85395c6e914dd5b321a26c61362/plugin/src/test/java/io/jenkins/plugins/casc/JenkinsConfigTest.java#L25-L28